src/pudl_archiver/depositors/fsspec.py

```diff
@@ -0,0 +1,272 @@
+    api_client: FsspecAPIClient
+    dataset_id: str
+
+    async def list_files(self):
+        """List files."""
+        return self.deposition.file_list
+
+    def get_deposition_link(self) -> str:
+        """Return link to deposition."""
+        return self.deposition.deposition_path.as_uri()
+
+    async def get_file(self, filename: str) -> bytes:
+        """Download file from deposition."""
+        return self.api_client.get_file(self.deposition, filename)
+
+    async def open_draft(self) -> "FsspecDraftDeposition":
+        """Open a new draft to make edits."""
+        return FsspecDraftDeposition(
+            deposition=self.deposition,
+            settings=self.settings,
+            api_client=self.api_client,
+            dataset_id=self.dataset_id,
+        )
+
+
+class FsspecDraftDeposition(DraftDeposition):
+    """Represents draft version of fsspec deposition."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    settings: RunSettings
+    deposition: Deposition
+    api_client: FsspecAPIClient
+    dataset_id: str
+
+    async def list_files(self):
+        """List files."""
+        return self.deposition.file_list
+
+    def get_deposition_link(self) -> str:
+        """Return link to deposition."""
+        return self.deposition.deposition_path.as_uri()
+
+    async def publish(self) -> FsspecPublishedDeposition:
+        """Publish deposition."""
+        return FsspecPublishedDeposition(
+            dataset_id=self.dataset_id,
+            api_client=self.api_client,
+            settings=self.settings.model_copy(update={"initialize": False}),
+        )
+
+    async def create_file(
+        self,
+        filename: str,
+        data: BinaryIO,
+    ) -> "FsspecDraftDeposition":
+        """Create a file in a deposition."""
+        with (self.deposition.deposition_path / filename).open(mode="wb") as f:
+            f.write(data.read())
+
+        return self.model_copy(
+            update={
+                "deposition": Deposition.from_upath(self.deposition.deposition_path)
+            }
+        )
+
+    async def delete_file(self, filename: str) -> "FsspecDraftDeposition":
+        """Delete a file from a deposition."""
+        (self.deposition.deposition_path / filename).unlink()
+        return self.model_copy(
+            update={
+                "deposition": Deposition.from_upath(self.deposition.deposition_path),
+            }
+        )
+
+    async def get_file(self, filename: str) -> bytes:
+        """Download file from deposition."""
+        return self.api_client.get_file(self.deposition, filename)
+
+    async def cleanup_after_error(self, e: Exception):
+        """Cleanup draft after an error during an archive run."""
+        logger.error(
+            f"Failed while creating new deposition: {traceback.print_exception(e)}"
+        )
+
+    async def delete_deposition(self) -> None:
+        """Delete an un-submitted deposition."""
+        pass
```

Comment:
Do we want to do something more forceful than passing here? E.g. log or raise an error? Or actually delete the files in the folder? In what context is this getting called?

Reply:
Right now this will only get called if no changes are detected after attempting to create a new archive. Since this backend doesn't support versioning this doesn't really make sense, so I've changed this to raise an error.